runtime/doc/if_lua.txt

```diff
@@ -293,7 +353,76 @@ vim.types						*lua-vim.types*
 	values corresponding to `vim.types.float`, `vim.types.array` and 
 	`vim.types.dictionary` will not change or that `vim.types` table will 
 	only contain values for these three types.
+------------------------------------------------------------------------------
+vim.helpers.* helper functions
+
+This modules provides lua-only helper functions to assist in writing lua
+plugins.
+
+vim.helpers.split({s}, {sep}, {plain})                       *vim.helpers.split*
+	Split a given string by a separator. Returns a table containing the 
+	splitted components. The separator can be a lua pattern, see
+		    https://www.lua.org/pil/20.2.html
+	Setting {plain} to `true` turns of pattern matching, as it is passed
+	to `string:find`, see 
+		http://lua-users.org/wiki/StringLibraryTutorial
+
+	Parameters:~
+		{s}     String: String to split
+		{sep}   String: Separator pattern. If empty, split by chars.
+		{plain} Boolean: If false, match {sep} verbatim
+
+	Return:~
+		Table of strings, which are the components of {s} after
+		splitting	
+
+vim.helpers.gsplit({s}, {sep}, {plain})                     *vim.helpers.gsplit*
+	Split a given string by a separator. Returns an iterator of the 
+	splitted components. The separator can be a lua pattern, see
+		    https://www.lua.org/pil/20.2.html
+	Setting {plain} to `true` turns of pattern matching, as it is passed
+	to `string:find`, see 
+		http://lua-users.org/wiki/StringLibraryTutorial
+
+	Parameters:~
+		{s}     String: String to split
+		{sep}   String: Separator pattern. If empty, split by chars.
+		{plain} Boolean: If false, match {sep} verbatim
+
+	Return:~
+		Iterator of strings, which are the components of {s} after
+		splitting	
+
+vim.helpers.inspect({object}, {options})                   *vim.helpers.inspect*
```

Comment:
Most functions should live directly on `vim.`, not a sub-module `vim.helpers`. We own the `vim.*` module; the only "forbidden" names are those used by legacy `if_lua`:

```
vim
:lua for k,v in pairs(vim) do print(k) end

buffer
open
window
lastline
firstline
type
line
eval
dict
beep
list
command
```